src/catchup/ApplyBucketsWork.cpp

```diff
@@ -192,8 +225,19 @@ ApplyBucketsWork::onRun()
         return State::WORK_RUNNING;
     }
 
-    CLOG(DEBUG, "History") << "ApplyBuckets : done, restarting merges";
-    mApp.getBucketManager().assumeState(mApplyState, mMaxProtocolVersion);
+    CLOG(INFO, "History") << "ApplyBuckets : done, restarting merges";
+    // Use old-style merges from HAS
+    bool useHASFutureBuckets =
```

Comment:
`useHASFutureBuckets` gets calculated by examining `mMaxProtocolVersion` here, but it's then passed to `BucketManager::assumeState` and on to `BucketList::restartMerges`. In that function, it's used as one half of an OR-condition, the other half of which (`removeShadows`) comes from `spillLevelWithoutShadows`, which is itself also derived by examining `mMaxProtocolVersion`. So this _seems_ redundant to me: the condition decided by `useHASFutureBuckets` is being partially re-decided in `BucketList::spillLevelWithoutShadows` and both decisions are OR'ed together and only used in that one condition in `BucketList::restartMerges`. I think one of the two calculations should go, and I _think_ it's this flag? Not sure though.